### PR TITLE
Mario/2858 add warning field backend

### LINKED ISSUE #2858: Add warning field to backend
**Is your feature request related to a problem? Please describe.**

We want to inform users quick, if the backend shows bad behavior. We currently need to update Lunie to do so.

**Describe the solution you'd like**

Add a field to our backend (Hasura) where we can set a warning message. Lunie queries this field and displays the warning bar.

**Describe alternatives you've considered**

Only use Twitter for warnings. Update Lunie manually with the warnings.

**Additional context**

### PR DESCRIPTION
**Pull request by mariopino**
Closes #2858

**Description:**

When a row is found in maintenance table the message is displayed to the user.

Maintenance table in Hasura backend have two columns: message and type. 

Type can be ok, warning, danger or empty (default) and changes the maintenance bar background color accordingly to each type.

Thank you! 🚀

---

For contributor:

- [ ] Added changes entries. Run `yarn changelog` for a guided process.
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

**Comment by faboweb** [comment]
@mariopino pending changelog missing

**Comment by mariopino** [comment]
> @mariopino pending changelog missing

Done!

File: src/components/common/MaintenanceBar.vue [hunk 1]
```diff
@@ -49,6 +75,18 @@ export default {
   color: var(--bright);
 }
 
+.maintenance-bar.ok {
+  background-color: #61b360;
```
Comment (faboweb) [inline src/components/common/MaintenanceBar.vue]: please use colors from variables that already exist or add the colors to the variables. This way we can have an overview about used colors.

File: src/components/common/MaintenanceBar.vue [hunk 2]
```diff
@@ -1,23 +1,36 @@
...
-    <p>
-      We've identified problems with our servers that are causing issues for
-      some of our users. We apologize for the disruption and are working on a
-      fix.
-    </p>
-    <a class="close">
-      <i class="material-icons" @click="close">close</i>
-    </a>
+  <div>
+    <div v-if="session.maintenanceBar && show" class="maintenance-bar">
```
Comment (faboweb) [inline src/components/common/MaintenanceBar.vue]: Do we still need this switch? This should now be obsolete as we show the maintenance bar only of there are records in the DB?

File: src/components/common/MaintenanceBar.vue [hunk 3]
```diff
@@ -1,23 +1,36 @@
...
+      <p>
+        We've identified problems with our servers that are causing issues for
+        some of our users. We apologize for the disruption and are working on a
+        fix.
+      </p>
+      <a class="close">
+        <i class="material-icons" @click="close">close</i>
+      </a>
+    </div>
+    <div v-if="maintenance.length > 0 && show" class="maintenance-bar" v-bind:class="maintenance[0].type">
```
Comment (faboweb) [inline src/components/common/MaintenanceBar.vue]: Why did you limit this just one record? Would it be possible to show multiple messages?

File: src/components/common/MaintenanceBar.vue (hunk 2 above)
Comment (faboweb) [reply src/components/common/MaintenanceBar.vue]: This is also replicating the same html structure twice. I propose we remove this section completely. Remember to also remove the `session.maintenanceBar` in the store

File: src/components/common/MaintenanceBar.vue (hunk 3 above)
Comment (mariopino) [reply src/components/common/MaintenanceBar.vue]: > Why did you limit this just one record? Would it be possible to show multiple messages?

Sure, I'll check the needed css so we can stack multiple messages.

**Comment by mariopino** [comment]
- Obsolete maintenance bar code removed.

- Removed session.maintenanceBar from store.

- Multiple messages supported.

- Now possible message types are:

  - **success**  background-color: var(--success)
  - **warning** background-color: var(--warning)
  - **danger** background-color: var(--danger)

- Also there is a new field in maintenance table in Hasura backend: show (boolean), which let us have predefined messages and enable them on demand.

**Comment by mariopino** [comment]
Pending to write some tests to cover the new behavior of the component mocking apollo query data ...

**Comment by faboweb** [comment]
> Also there is a new field in maintenance table in Hasura backend: show (boolean), which let us have predefined messages and enable them on demand.

Sweet idea.

File: src/components/common/MaintenanceBar.vue [hunk 4]
```diff
@@ -100,11 +100,11 @@ export default {
     justify-content: space-around;
   }
 
-  .maintenance-bar:nth-child(2) {
+  div > div.maintenance-bar:nth-child(2) {
```
Comment (jbibla) [inline src/components/common/MaintenanceBar.vue]: hmm. why is this necessary @mariopino ?

the css is `scoped` - seems like we could accomplish this somehow with less specificity

Comment (mariopino) [reply src/components/common/MaintenanceBar.vue]: In screen sizes with max-width less than 1025px, maintenance-bar messagess are shown in the bottom of the page, but they are not stacked, so only the last message is displayed.

So that css code is needed to stack the messages at the bottom. I'm not happy with the solution thought, sure must be a better way to acomplish that.

In the case we want to keep the current behavior (only the last message is displayed in screens <= 1024px width) we can simply delete those css statements.

Comment (jbibla) [reply src/components/common/MaintenanceBar.vue]: ok cool. i understand. 

i think you are right in trying to solve this stacking problem. but i think we should find a solution that doesn't involve `div > div`. perhaps class names for each banner? or some kind of architectural change might be in order.

Comment (mariopino) [reply src/components/common/MaintenanceBar.vue]: @jbibla  css was updated in the lasts commits so now it's:

```
 .message-2 {
    bottom: 36px;
  }
  .message-3 {
    bottom: 72px;
  }
  .message-4 {
    bottom: 108px;
  }
  .message-5 {
    bottom: 144px;
  }
```

I would like to use scss/vars to increment the bottom offset automatically in each message but I don't know if it is possible.

Comment (jbibla) [reply src/components/common/MaintenanceBar.vue]: i don't think we have scss at the moment. i think what you're trying to resolve is a symptom of a bigger issue.

i'll take a look now and try a couple things out.

Comment (faboweb) [reply src/components/common/MaintenanceBar.vue]: I am probably wrong, but can't you put the messages in a container and then just position them relative to each other?

Comment (jbibla) [reply src/components/common/MaintenanceBar.vue]: no you're correct. mario and i talked about this yesterday. it will require some modifications. 

i opened https://github.com/luniehq/lunie/issues/2967 as a separate issue.
